src/pso/swarm.rs

```diff
@@ -30,24 +31,23 @@ impl Swarm {
     }
 
     pub fn update_velocities(&mut self, inertia_weight: &f64, cognitive_coefficient: &f64, social_coefficient: &f64) {
-        for i in 0..self.particles.len() {
-            self.particles[i].update_velocity(&self.best_position, inertia_weight, cognitive_coefficient, social_coefficient);
-        }
+        self.particles.par_iter_mut()
```

Comment:
I believe we can leave it for now, but  I think we should left parallelization out until we have some benchmarking system implemented. Moreover we should discuss introduction of parallelism with our client, to discuss whether there are any expectations.